engine/src/main/java/com/datatorrent/stram/engine/GenericNode.java

```diff
@@ -394,6 +463,15 @@ else if (tracker.ports[trackerIndex] == activePort) {
               case END_STREAM:
                 activePort.remove();
                 buffers.remove();
+                if (firstWindowId == -1) {
+                  // this is for recovery from a checkpoint for DelayOperator
+                  if (delay) {
+                    // if it's a DelayOperator and this is the first RESET_WINDOW (start) or END_STREAM (recovery),
+                    // fabricate the first window
+                    fabricateFirstWindow((Operator.DelayOperator)operator, windowAhead);
+                  }
+                  firstWindowId = t.getWindowId();
```

Comment:
Again same Should this firstWindowId not be assigned when delay is true and firstWindowId == -1?